docs/source/internals/engine.rst

```diff
@@ -6,6 +6,15 @@
     Engine
     ******
 
+    .. _internal_architecture:engine:caching:
+
+    Controlling caching
+    -------------------
+
+    Old Docs: This should also contain the contents of "Controlling caching" inside of developer_guide/core/caching.rst.
+    You should include links to the other sections topics:provenance:caching: and how-to:run-codes:caching:, and in those sections as well.
+    Example for topics:provenance:caching: If you want more details of the internal architecture instead, you can check out the :ref:`relevant subsection <internal_architecture:engine:caching>` in the engine documentation.
```

Comment:
The engine part was being done by @chrisjsewell and a reminder to do this was already added to that issue. I added this here mostly because since we already had the "duplicated" Engine task (meaning, both the issue and the todo in the files), we might as well have all the info in both places. If I copy it myself it might conflict with the other content of the section and be annoying to resolve, not to mention that it might already have been included in that work.

Reply:
If he is writing and improving the caching documentation, wouldn't it then make sense to have him also move/incorporate these parts, so he can make sure the whole thing is coherent?

Reply:
Not the caching in particular, he is writing essentially from scratch the docs for the internals of the engine, which were lower priority than trying to get all of the old documentation back somewhere in the new and hence he didn't need to finish this for the release. What we did want was to try to get back as much of the old docs as possible, and since the "topics" section was already present, moving all of the "user-facing" content of caching there seemed like a possibility. If you think is better to wait to emphasize on the consistency over having available these sections on caching/hashing now for the release, then I guess I can block this until the engine section is finished.